var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py

```diff
@@ -59,13 +65,31 @@ def headers(self):
         include_path = join_path(self.component_path, 'include')
         return find_headers('*', include_path)
 
+    # provide cluster libraries if +cluster variant is used or
+    # the scalapack virtual package was requested
+    def cluster(self):
+        return '+cluster' in self.spec or 'scalapack' in self.spec
```

Comment:
I think this can just be `return '+cluster' in self.spec`. As it is, you can't check if a spec X provides a virtual by evaluating `virtual in x`: you'd have to say `x.satisfies(virtual)`. Since this `provides(scalapack, when="+cluster")`, you can just check if `+cluster in self.spec`.

Reply:
Fixed